src/components/containers/EditStudentContainer.js

```diff
@@ -0,0 +1,225 @@
+                if (this.isBlank(this.state.email)) {
+                    emailError = true;
+                }
+                if (this.isBlank(this.state.lastname)) {
+                    lastnameError = true;
+                }
+                break;
+            case "lastname":
+                if (this.isBlank(event.target.value)) {
+                    lastnameError = true;
+                }
+                if (this.isBlank(this.state.firstname)) {
+                    firstnameError = true;
+                }
+                if (this.isBlank(this.state.email))
+                    emailError = true;
+                break;
+            case "email":
+                if (this.isBlank(event.target.value))
+                    emailError = true;
+                if (this.isBlank(this.state.firstname))
+                    firstnameError = true;
+                if (this.isBlank(this.state.lastname))
+                    lastnameError = true;
+                break;
+            default:
+                return;
+        }
+
+        this.setState(prevstate => ({
+            errors: {
+                ...prevstate.errors,
+                firstname: firstnameError,
+                lastname: lastnameError,
+                email: emailError
+            }
+        }));
+    }
+
+    /* Validate Entire Form */
+  validateForm() {
+    let firstnameError = false;
+    let lastnameError = false;
+    let emailError = false;
+
+    if (this.isBlank(this.state.firstname))
+        firstnameError = true;
+
+    if (this.isBlank(this.state.lastname))
+        lastnameError = true;
+
+    if (this.isBlank(this.state.email))
+        emailError = true;
+
+    if (firstnameError || lastnameError || emailError) {
+        this.setState(prevstate => ({
+            errors: {
+                ...prevstate.errors,
+                firstname: firstnameError,
+                lastname: lastnameError,
+                email: emailError,
+                validate: true
+            }
+        }))
+        return false;
+    }
+    else {
+        this.setState(prevstate => ({
+            errors: {
+                ...prevstate.errors,
+                firstname: firstnameError,
+                lastname: lastnameError,
+                email: emailError
+            }
+        }));
+        return true;
+    }
+}
+
+/* Submit Form */
+handleSubmit = async event => {
+    event.preventDefault();
+    const path = window.location.pathname;
+    const isEdit = path.includes("edit");
+    const id = parseInt(path.split('/').slice(-2, -1)[0]);
+
+
+    if (!this.validateForm()) {
+        return;
+    }
+
+    const shouldUseImageUrl = (this.state.imageUrl && this.state.imageUrl.trim() !== "") ? true : false;
+
+    let student = {
+        firstname: this.state.firstname,
+        lastname: this.state.lastname,
+        email: this.state.email,
+        ...(shouldUseImageUrl && { imageUrl: this.state.imageUrl }),
+        id: isEdit ? id : null,
+    }
```

Comment:
Currently, need to add a `gpa` field so the gpa also updates when editing student.

When you implement the `campus` dropdown, you'll also need to add that to the student object.